MotionMark/resources/statistics.js

```diff
@@ -249,17 +253,22 @@ Regression = Utilities.createClass(
             };
         }
 
+        // Sort by increasing complexity.
+        var sortedSamples = samples.slice().sort((a, b) => a[complexityIndex] - b[complexityIndex]);
+        
         // x is expected to increase in complexity
-        var iterationDirection = endIndex > startIndex ? 1 : -1;
-        var lowComplexity = getComplexity(samples, startIndex);
-        var highComplexity = getComplexity(samples, endIndex);
+        var lowComplexity = sortedSamples[0][complexityIndex];
+        var highComplexity = sortedSamples[samples.length - 1][complexityIndex];
```

Comment:
I am confused here. I understand the individual ramps are sorted in descending order based on the complexity. But I do not understand how this code can work without sorting when calculating the regression for all the ramps data when is collected in `RampController.processSamples()`.

Reply:
I'm not sure I understand; `sortedSamples` is sorted a few lines above.

Reply:
Sorry for the confusion. Without this PR, how do these two lines return min/max complexity if samples is not sorted?
```
       var lowComplexity = getComplexity(samples, startIndex);
       var highComplexity = getComplexity(samples, endIndex);
```
Does `lowComplexity` mean the complexity at the beginning of sampling and it is not necessarily the minimum complexity?

Reply:
Right, this was why this class was confusing. It's up to the caller to pass `startIndex`, `endIndex` in the order which results in increasing complexity, which might result in the need to iterate backwards, hence `iterationDirection`.